sdk/samples/iot/provisioning/src/paho_iot_provisioning_example.c

```diff
@@ -230,84 +231,77 @@ static az_result read_configuration_entry(
   }
   else
   {
-    (void)printf("(missing) Please set the %s environment variable.\n", env_name);
+    LOG_ERROR("(missing) Please set the %s environment variable.\n", env_name);
     return AZ_ERROR_ARG;
   }
 
   return AZ_OK;
 }
 
-static int connect_device()
+static void connect_client_to_provisioning_service()
 {
   int rc;
 
-  MQTTClient_SSLOptions mqtt_ssl_options = MQTTClient_SSLOptions_initializer;
-  MQTTClient_connectOptions mqtt_connect_options = MQTTClient_connectOptions_initializer;
-
-  // NOTE: We recommend setting clean session to false in order to receive any pending messages
-  mqtt_connect_options.cleansession = false;
-  mqtt_connect_options.keepAliveInterval = AZ_IOT_DEFAULT_MQTT_CONNECT_KEEPALIVE_SECONDS;
-
-  // Get the MQTT username used to connect to the Provisioning Service
   if (az_failed(
           rc = az_iot_provisioning_client_get_user_name(
-              &provisioning_client, mqtt_username, sizeof(mqtt_username), NULL)))
+              &provisioning_client, mqtt_username_buffer, sizeof(mqtt_username_buffer), NULL)))
   {
-    (void)printf("Failed to get MQTT username: az_result return code 0x%04x .\n", rc);
-    return rc;
+    LOG_ERROR("Failed to get MQTT username: az_result return code 0x%04x .\n", rc);
+    exit(rc);
   }
 
-  // This sample uses X509 authentication so the password field is set to NULL
-  mqtt_connect_options.username = mqtt_username;
-  mqtt_connect_options.password = NULL;
+  MQTTClient_connectOptions mqtt_connect_options = MQTTClient_connectOptions_initializer;
+  mqtt_connect_options.username = mqtt_username_buffer;
+  mqtt_connect_options.password = NULL; // This sample uses x509 authentication.
+  mqtt_connect_options.cleansession = false; // Set to false so can receive any pending messages.
+  mqtt_connect_options.keepAliveInterval = AZ_IOT_DEFAULT_MQTT_CONNECT_KEEPALIVE_SECONDS;
 
-  // Set the device cert for tls mutual authentication
-  mqtt_ssl_options.keyStore = (char*)x509_cert_pem_file;
-  if (*x509_trust_pem_file != '\0')
+  MQTTClient_SSLOptions mqtt_ssl_options = MQTTClient_SSLOptions_initializer;
+  mqtt_ssl_options.keyStore = (char*)x509_cert_pem_file_buffer;
```

Comment:
humm??

```suggestion
  mqtt_ssl_options.keyStore = (char*)x509_cert_pem_filepath_buffer;
```

Reply:
I filepath is more descriptive.  I can add that to both of these:

// Buffers
static char x509_cert_pem_file_buffer[256] = { 0 };
static char x509_trust_pem_file_buffer[256] = { 0 };